File: Study Note/Algorithm Problems/SWEA/D3/5515 2016년 요일 맞추기 (pass)/SWEA_5515.c
#include <stdio.h>
int main(void)
{
	int test_case;
	int T, m, d, i;
  int month[13] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
  int week[7] = {3, 4, 5, 6, 0, 1, 2};
	scanf("%d", &T);
	for (test_case = 1; test_case <= T; ++test_case)
	{
		scanf("%d %d", &m, &d);
		for (i = 1; i < m; i++) d += month[i];
    printf("#%d %d\n", test_case, week[d%7]);
	}
	return 0; 
}

File: Study Note/Algorithm Problems/SWEA/D3/5515 2016년 요일 맞추기 (pass)/SWEA_5515.py
import sys
sys.stdin = open("input.txt", "r")

month = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
week = [3, 4, 5, 6, 0, 1, 2]
T = int(input())
for test_case in range(1, T + 1):
    # ///////////////////////////////////////////////////////////////////////////////////
    '''

        이 부분에 여러분의 알고리즘 구현이 들어갑니다.

    '''
    # ///////////////////////////////////////////////////////////////////////////////////
    m, d = map(int, input().split())
    days = d
    for i in range(1, m):
        days += month[i]
    print('#{} {}'.format(test_case, week[days % 7]))